### PR TITLE
README: Adjust Linux and macOS support platform and architecture

### PR DESCRIPTION
**Pull request by tesuji**
cc #74163

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -211,11 +211,11 @@ fetch snapshots, and an OS that can execute the available snapshot binaries.
...
 
-| Platform / Architecture    | x86 | x86_64 |
-|----------------------------|-----|--------|
-| Windows (7, 8, 10, ...)    | ✓   | ✓      |
-| Linux (2.6.18 or later)    | ✓   | ✓      |
-| macOS (10.7 Lion or later) | ✓   | ✓      |
+| Platform / Architecture                     | x86 | x86_64 |
+|---------------------------------------------|-----|--------|
+| Windows (7, 8, 10, ...)                     | ✓   | ✓      |
+| Linux (kernel 2.6.32, glibc 2.11 or later)  | ✓   | ✓      |
```
Comment (pickfire) [inline README.md:217]: How about musl libc?

Comment (pickfire) [reply README.md:217]: True, as of the moment it is tier 2.

File: README.md [hunk 2]
```diff
@@ -211,11 +211,11 @@ fetch snapshots, and an OS that can execute the available snapshot binaries.
...
+| macOS (10.7 Lion or later)                  | ✓   | ✓      |
```
Comment (cuviper) [inline README.md:218]: While we're here, can you remove the macOS x86 check? ([ref](https://blog.rust-lang.org/2020/01/03/reducing-support-for-32-bit-apple-targets.html))

Comment (lzutao) [reply README.md]: Done.

**Pull request merged**